proto/src/main/proto/pb.proto

```diff
@@ -769,7 +769,7 @@ message Filter {
     bool disable_api = 27;
     bool disable_mempool_validation = 28;
     bool disable_pow_message = 29;
-    int32 pow_difficulty = 30;
+    double pow_difficulty = 30;
```

Comment:
That would possibly be OK. But on a fast CPU running the PoW search in parallel (if we made an enhancement to do that), it might take around 1 microsecond per unit difficulty or less, so a difficulty set to INT_MAX would correspond to an average search time of less that 20 minutes on a fast machine. I'm not sure we would ever want to set the difficulty that high in case of heavy DoS attack, but it would still make posting BSQ swap offers just barely usable for people with fast machines.

Reply:
Ah ok. Yes then we should not limit it by choosing a int type. What about a long type? Would it hurt anything on the Equihash side? But no strong opinion, just was wondering why a double is used.

Reply:
I was attempting to base it loosely on the Bitcoin difficulty, which is floating point. E.g. in the RPC source code of Bitcoin Core, https://github.com/bitcoin/bitcoin/blob/master/src/rpc/blockchain.cpp, they internally use a double to represent difficulty.